packages/react-components/src/components/VideoGallery.tsx

```diff
@@ -119,15 +147,16 @@ export const VideoGallery = (props: VideoGalleryProps): JSX.Element => {
     styles,
     layout,
     onRenderAvatar,
-    showMuteIndicator
+    showMuteIndicator,
+    dominantSpeakers
   } = props;
   const [sortedRemoteParticipants, setSortedRemoteParticipants] = useState<VideoGalleryRemoteParticipant[]>([]);
```

Comment:
This should be `useRef` not `useState`.

You have not (correctly!) dropped the `useEffect` block, so this doesn't need to be a state variable.

State variables should track information that changes internal to this component *without any change from props*. That is not the case here. This is just caching some computed values from props.

Reply:
Could you point me to the source of this claim?
Replacing `useState` with `useRef` here will prevent the component from re-rendering when remote participants are sorted.
And hence, need a state variable.

Reply:
Heh: s/not/now -- I was saying that you have *correctly* dropped the `useEffect` block, but typo'ed to "not correctly". Classic :P

Reply:
Actually, my comment above is not fully correct. You don't need `useRef` either. `useMemo` already returns the same value for same inputs (so caching happens in a global context internal to `useMemo`). So you are good to go with just a local variable.

    const sortedRemoteParticipants = useMemo(() => sortParticipants(remoteParticipants, dominantSpeakers), [remoteParticipants, dominantSpeakers])

Reply:
Ok I understand what you mean now. Thanks, good suggestion.